sktime/forecasting/base/_sktime.py

```diff
@@ -589,9 +589,10 @@ def _set_fh(self, fh):
 class _BaseWindowForecaster(_SktimeForecaster):
     """Base class for forecasters that use """
 
-    def __init__(self, window_length=None):
+    def __init__(self, length_y, window_length=None):
         super(_BaseWindowForecaster, self).__init__()
-        self.window_length = window_length
+        self.length_y = len(length_y)
+        self.window_length = window_length * length_y
```

Comment:
Rather than passing y to the constructor (`__init__`), we would want to compute the window length when we call `split`

Reply:
Thank you @mloning.

Reply:
I am thinking of adding the n_timepoints as an argument in check_windows_length. This way, I would not have to edit all instances where `windows_length = check_windows_length()`